src/Interpreters/InterpreterSelectQueryAnalyzer.cpp

```diff
@@ -172,15 +172,21 @@ static QueryTreeNodePtr buildQueryTreeAndRunPasses(const ASTPtr & query,
 
 
 InterpreterSelectQueryAnalyzer::InterpreterSelectQueryAnalyzer(
-    const ASTPtr & query_,
-    const ContextPtr & context_,
-    const SelectQueryOptions & select_query_options_,
-    const Names & column_names)
+    const ASTPtr & query_, const ContextPtr & context_, const SelectQueryOptions & select_query_options_, const Names & column_names)
     : query(normalizeAndValidateQuery(query_, column_names))
     , context(buildContext(context_, select_query_options_))
     , select_query_options(select_query_options_)
     , query_tree(buildQueryTreeAndRunPasses(query, select_query_options, context, nullptr /*storage*/))
     , planner(query_tree, select_query_options)
+    , query_plan_with_parallel_replicas_builder(
```

Comment:
Let's do it only if the feature is enabled. We can create many Interpreters during a query execution

Reply:
> Let's do it only if the feature is enabled.

This lambda is called from the optimisation code, and only if the optimisation is enabled. So, AFAIU, it already works this way.

Reply:
It looks we're checking it but when applying optimization